addon/adapters/adapter.js

```diff
@@ -156,6 +173,46 @@ export default Adapter.extend(ImportExportMixin, {
     return null;
   },
 
+  _queryFilter(record, query = {}) {
+    return keys(query).every((key) => {
+      const value = query[key],
+        matches = key.match(/(.*)Id$/),
```

Comment:
Not sure I like this way of querying for relationships, there could be possible conflicts if someone names attributes to match `/(.*)Id$/`. Could we go in keeping with `filter: {}` and use something like `filter: { relationships: { id: '123', type: 'user' } }`. It would be useful for polymorphic relationships, when using a regexp for type or omitting one altogether.